### PR TITLE
fix: linter tests

### PR DESCRIPTION
**Pull request by wep21**
# 🦟 Bug fix

Fixes #<NUMBER>

## Summary

fix some linter tests

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: ros_ign_gazebo/CMakeLists.txt [hunk 1]
```diff
@@ -81,7 +81,15 @@ install(TARGETS
...
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_cpplint
+  )
   ament_lint_auto_find_test_dependencies()
+
+  find_package(ament_cmake_cpplint)
+  ament_cpplint(
+    FILTERS -build/include_order
```
Comment (wep21) [inline ros_ign_gazebo/CMakeLists.txt]: ignore include order once due to https://github.com/ament/ament_lint/pull/374.

**Comment by wep21** [comment]
@chapulina I appreciate it if you do bloom-release to `rolling/humble` distribution after merging this PR.

File: .github/workflows/build-and-test.sh [hunk 2]
```diff
@@ -13,7 +13,7 @@ apt install -qq -y lsb-release wget curl build-essential
 # Citadel, Edifice and Fortress get Ignition with rosdep for Focal
 
 # Dependencies.
-echo "deb http://packages.ros.org/ros2-testing/ubuntu `lsb_release -cs` main" > /etc/apt/sources.list.d/ros2-testing.list
+echo "deb http://packages.ros.org/ros2/ubuntu `lsb_release -cs` main" > /etc/apt/sources.list.d/ros2.list
```
Comment (chapulina) [inline .github/workflows/build-and-test.sh]: We changed to `testing` on purpose, it's best practice for CI, see https://github.com/gazebosim/ros_gz/pull/246#issuecomment-1106783015

**Review by chapulina — APPROVED** [review]
Thanks for the PR! I pushed 94da01dde3ddbc7082be81c9e083bf56859aab81 with 2 changes:

* Restore testing repo for CI
* Instead of suppressing the ament warnings, fix the include headers. This is finally our chance to group the Gazebo and ROS C++ headers together 😄 

I noticed there's a test failure both locally for me and on CI. I'm not sure if it's something we're doing wrong or some regression upstream. I'll merge with that failure for now, since this PR is about linters.

```
[4.654s] 1: [test_ros_subscriber-3] cannot publish data, at ./src/rmw_publish.cpp:62 during '__function__'
[4.655s] 1: [test_ros_subscriber-3] [ERROR] [1653006329.567666397] [ros_subscriber.rclcpp]: Error in destruction of rcl subscription handle: Failed to delete datareader, at ./src/subscription.cpp:52, at ./src/rcl/subscription.c:183
[4.655s] 1: [test_ros_subscriber-3] cannot publish data, at ./src/rmw_publish.cpp:62 during '__function__'
[4.655s] 1: [test_ros_subscriber-3] Fail in delete datareader, at ./src/rmw_service.cpp:103 during '__function__'
```

File: ros_ign_gazebo/CMakeLists.txt (hunk 1 above)
Comment (chapulina) [reply ros_ign_gazebo/CMakeLists.txt]: I remove these suppressions and instead moved the C++ headers to make the linter pass.

**Pull request merged**